ts/interactions/conversations/unsendingInteractions.ts

```diff
@@ -339,24 +339,36 @@ export async function deleteMessagesByIdForEveryone(
   const messageCount = selectedMessages.length;
   const moreThanOne = messageCount > 1;
 
-  window.inboxStore?.dispatch(
-    updateConfirmModal({
-      title: window.i18n('deleteForEveryone'),
-      message: moreThanOne
-        ? window.i18n('deleteMessagesQuestion', [messageCount.toString()])
-        : window.i18n('deleteMessageQuestion'),
-      okText: window.i18n('deleteForEveryone'),
-      okTheme: SessionButtonColor.Danger,
-      onClickOk: async () => {
-        await doDeleteSelectedMessages({ selectedMessages, conversation, deleteForEveryone: true });
-
-        // explicity close modal for this case.
-        window.inboxStore?.dispatch(updateConfirmModal(null));
-        return;
-      },
-      closeAfterInput: false,
-    })
-  );
+  if (window.getSettingValue('confirm-deletions')) {
+    window.inboxStore?.dispatch(
+      updateConfirmModal({
+        title: window.i18n('deleteForEveryone'),
+        message: moreThanOne
+          ? window.i18n('deleteMessagesQuestion', [messageCount.toString()])
+          : window.i18n('deleteMessageQuestion'),
+        okText: window.i18n('deleteForEveryone'),
+        okTheme: SessionButtonColor.Danger,
+        onClickOk: async () => {
+          await doDeleteSelectedMessages({
+	    selectedMessages,
+	    conversation,
+	    deleteForEveryone: true
+	  });
+
+          // explicity close modal for this case.
+          window.inboxStore?.dispatch(updateConfirmModal(null));
+          return;
+        },
+        closeAfterInput: false,
+      })
+    );
+  } else {
+    await doDeleteSelectedMessages({
+      selectedMessages,
+      conversation,
+      deleteForEveryone: true
+    });
+  }
```

Comment:
create a temp function to avoid duplication
create something like this before the `window.getSettingValue('confirm-deletions'`
```
const doDeleteAction = async () => doDeleteSelectedMessages({
      selectedMessages,
      conversation,
      deleteForEveryone: true
    });
   ```
    and call it in line 351 and 365